src/frontend/js/components/CourseRunEnrollment/index.tsx

```diff
@@ -220,7 +230,7 @@ const CourseRunEnrollment: React.FC<CourseRunEnrollmentProps & CommonDataProps>
           </button>
           {step === Step.ENROLLMENT_FAILED ? (
             <div className="course-run-enrollment__errortext">
-              <FormattedMessage {...messages.enrollmentFailed} />
+              {error != undefined && error.code == 400 ? <>{error.message}</> : <FormattedMessage {...messages.enrollmentFailed} />}
```

Comment:
Use strict operators. Furthermore, you could use an optional chaining operator (`?`) instead to test if `error` is defined.

```suggestion
              {error?.code === 400 ? error.message : <FormattedMessage {...messages.enrollmentFailed} />}

```